iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/vector.inl

```diff
@@ -370,27 +359,45 @@ inline typename vector<T, Capacity>::iterator vector<T, Capacity>::erase(iterato
         }
         at(n).~T();
         m_size--;
+        if (m_size == 0)
+        {
+            return end();
+        }
+        if (index == 0)
+        {
+            return begin();
+        }
+        return &at(index - 1);
     }
     return nullptr;
 }
 
 template <typename T, uint64_t Capacity>
-T& vector<T, Capacity>::at_unchecked(const uint64_t index) noexcept
+inline T& vector<T, Capacity>::at_unchecked(const uint64_t index) noexcept
 {
     // AXIVION Next Construct AutosarC++19_03-A5.2.3 : const cast to avoid code duplication
     // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast)
     return const_cast<T&>(const_cast<const vector<T, Capacity>*>(this)->at_unchecked(index));
 }
 
 template <typename T, uint64_t Capacity>
-const T& vector<T, Capacity>::at_unchecked(const uint64_t index) const noexcept
+inline const T& vector<T, Capacity>::at_unchecked(const uint64_t index) const noexcept
 {
     // AXIVION Next Construct AutosarC++19_03-A5.2.4 : Type-safety ensured by template parameter
     // NOLINTJUSTIFICATION User accessible method at() performs bounds check
     // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast,cppcoreguidelines-pro-bounds-pointer-arithmetic)
     return reinterpret_cast<const T*>(m_data)[index];
 }
 
+template <typename T, uint64_t Capacity>
+inline void vector<T, Capacity>::clearFrom(const uint64_t startPosition) noexcept
+{
+    while (m_size > startPosition)
+    {
+        at_unchecked(--m_size).~T();
```

Comment:
Calling `pop_back` within this method would lead to unnecessary `m_size > 0` comparisons and we would discard the return value of `pop_back`. In my opinion the current implementation is the more efficient and elegant one but no hard feelings about it.

Reply:
@FerdinandSpitzschnueffler I'll leave it to you :)

Reply:
Then I'll leave it as it is ;)